pyintegration/deephaven2/_jcompat.py

```diff
@@ -56,4 +56,18 @@ def j_properties(d: Dict = None) -> jpy.JType:
         if value is None:
             value = ''
         r.setProperty(key, value)
-    return r
\ No newline at end of file
+    return r
+
+
+def j_map_to_dict(m):
+    """  Converts a java map to a python dictionary. """
+    if not m:
+        return None
+
+    r = {}
+    for e in m.entrySet().toArray():
+        k = e.getKey()
+        v = e.getValue()
+        r[k] = v
```

Comment:
I think this can be done with an explicit iterator to avoid array creation.  Efficiency change.

Reply:
I think jpy might handle it for some but not all types.